subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/validation/GradleMetadataValidationResolveIntegrationTest.groovy

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.validation
+
+import org.gradle.integtests.fixtures.GradleMetadataResolveRunner
+import org.gradle.integtests.fixtures.RequiredFeature
+import org.gradle.integtests.fixtures.RequiredFeatures
+import org.gradle.integtests.resolve.AbstractModuleDependencyResolveTest
+import org.gradle.test.fixtures.gradle.GradleFileModuleAdapter
+import spock.lang.Unroll
+
+class GradleMetadataValidationResolveIntegrationTest extends AbstractModuleDependencyResolveTest {
+
+    @RequiredFeatures([
+        @RequiredFeature(feature = GradleMetadataResolveRunner.GRADLE_METADATA, value = "true")
+    ])
+    def "can resolve if component gav information is missing"() {
+        GradleFileModuleAdapter.printComponentGAV = false
+        buildFile << """
+            repositories.all {
+                metadataSources {
+                    gradleMetadata()
+                }
+            }
+            dependencies {
+                conf 'org.test:projectA:1.1'
+            }
+        """
+
+        when:
+        repository {
+            'org.test:projectA:1.1'()
+        }
+        repositoryInteractions {
+            'org.test:projectA:1.1' {
+                expectResolve()
+            }
+        }
+
+        then:
+        succeeds ":checkDeps"
+        resolve.expectGraph {
+            root(":", ":test:") {
+                module("org.test:projectA:1.1")
+            }
+        }
+
+        cleanup:
+        GradleFileModuleAdapter.printComponentGAV = true
+    }
+
+    @RequiredFeatures([
+        @RequiredFeature(feature = GradleMetadataResolveRunner.GRADLE_METADATA, value = "true")
+    ])
+    @Unroll
+    def "fails with proper error if #label is not defined"() {
```

Comment:
I wonder about the value of this integration test. There is a matching unit test and so it seems like it only add build times and potential duplicate work if any of the covered conditions were to change in the future.
WDYT?

Reply:
I was thinking about this already. But in general our philosophy is to add integration tests for new behavior. (Sometimes - like here - it would be nice to have a easy way to add cheaper integration tests though...)

My reasoning is:
- The unit test is a replacement for a _schema_ (which we do not have) which defines which attributes are mandatory.
- The integration tests are for showing that a proper error message is presented to the user. Which was the original GH issue. If you run the integration tests against master, they would all still lead to failing builds, but with a bad error message (an NPE).

Reply:
Then maybe the `@Unroll` could be dropped and only one invalid case tested?

Reply:
:+1: I dropped the `@Unroll`.

Can you approve the PR if there are no other concerns from your side?

Reply:
Done!